src/serialize/json/denormalize.ts

```diff
@@ -0,0 +1,506 @@
+  return v !== undefined ? denormalizer.denormalize(v, ctx, path) : undefined
+}
+
+export class Factory {
+  makeForBom (ctx: { options: JSONDenormalizerOptions }): BomDenormalizer {
+    return new BomDenormalizer(this)
+  }
+
+  makeForMetadata (ctx: JSONDenormalizerContext): MetadataDenormalizer {
+    return new MetadataDenormalizer(this)
+  }
+
+  makeForComponent (ctx: JSONDenormalizerContext): ComponentDenormalizer {
+    return new ComponentDenormalizer(this)
+  }
+
+  makeForTool (ctx: JSONDenormalizerContext): ToolDenormalizer {
+    return new ToolDenormalizer(this)
+  }
+
+  makeForOrganizationalContact (ctx: JSONDenormalizerContext): OrganizationalContactDenormalizer {
+    return new OrganizationalContactDenormalizer(this)
+  }
+
+  makeForOrganizationalEntity (ctx: JSONDenormalizerContext): OrganizationalEntityDenormalizer {
+    return new OrganizationalEntityDenormalizer(this)
+  }
+
+  makeForHash (ctx: JSONDenormalizerContext): HashDenormalizer {
+    return new HashDenormalizer(this)
+  }
+
+  makeForLicense (ctx: JSONDenormalizerContext): LicenseDenormalizer {
+    return new LicenseDenormalizer(this)
+  }
+
+  makeForSWID (ctx: JSONDenormalizerContext): SWIDDenormalizer {
+    return new SWIDDenormalizer(this)
+  }
+
+  makeForExternalReference (ctx: JSONDenormalizerContext): ExternalReferenceDenormalizer {
+    return new ExternalReferenceDenormalizer(this)
+  }
+
+  makeForAttachment (ctx: JSONDenormalizerContext): AttachmentDenormalizer {
+    return new AttachmentDenormalizer(this)
+  }
+
+  makeForProperty (ctx: JSONDenormalizerContext): PropertyDenormalizer {
+    return new PropertyDenormalizer(this)
+  }
+
+  makeForUrl (ctx: JSONDenormalizerContext): UrlDenormalizer {
+    return new UrlDenormalizer(this)
+  }
+
+  makeForBomRef (ctx: JSONDenormalizerContext): BomRefDenormalizer {
+    return new BomRefDenormalizer(this)
+  }
+}
+
+interface JsonDenormalizer<TModel, TNormalized> {
+  denormalize: (data: TNormalized, ctx: JSONDenormalizerContext, path: PathType) => TModel | undefined
+}
+
+abstract class BaseJsonDenormalizer<TModel, TNormalized = Record<string, unknown>> implements JsonDenormalizer<TModel, TNormalized> {
+  protected readonly _factory: Factory
+
+  constructor (factory: Factory) {
+    this._factory = factory
+  }
+
+  get factory (): Factory {
+    return this._factory
+  }
+
+  abstract denormalize (data: TNormalized, ctx: JSONDenormalizerContext, path: PathType): TModel
+}
+
+export class BomDenormalizer extends BaseJsonDenormalizer<Models.Bom> {
+  denormalize (data: Record<string, unknown>, { options }: { options: JSONDenormalizerOptions }, path: PathType): Models.Bom {
+    assertEnum(data.bomFormat, ['CycloneDX'], [...path, 'bomFormat'])
+    assertEnum(data.specVersion, Object.keys(SpecVersionDict), [...path, 'specVersion'])
+    const spec = SpecVersionDict[data.specVersion as keyof typeof SpecVersionDict] as Protocol
+    if (!spec.supportsFormat(Format.JSON)) {
```

Comment:
:x: NO!
the (de)normalizer does not decide whether to run or not - it just does the job.
the (de)serializer might decide to not run because of constraints.